src/main/java/Duke.java

```diff
@@ -1,10 +1,146 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+/**
+ * Represents the main class Duke.
+ */
 public class Duke {
-    public static void main(String[] args) {
+    private static Ui ui = new Ui();
+    private static TaskList tasks = new TaskList();
+    private static Parser parser = new Parser();
+    private static Storage storage = new Storage();
+    private static String user_input;
+
+    /**
+     * Prints out the welcome message.
+     */
+    public static void show_welcome_msg() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("____________________________________________________________\n" +
+
+                " Hello! I'm Duke\n" +
+                " What can I do for you?\n" +
+                "____________________________________________________________\n");
+    }
+
+    /**
+     * Prints out the bye message.
+     */
+    public static void show_exit_msg() {
+        System.out.println("Bye. Hope to see you again soon!\n");
+    }
+
+    /**
+     * Ensembles all methods that each user-command requires.
+     * @param input user input
+     * @param c class command
+     * @throws InvalidCommandException
+     * @throws EmptyInputException
+     * @throws InvalidEventException
+     * @throws InvalidDeadlineException
+     */
+    public static void executeCommand(String input, Command c) throws InvalidCommandException, EmptyInputException, InvalidEventException, InvalidDeadlineException {
+        switch (c) {
+            case LIST:
+                tasks.listTaskMsg();
+                tasks.enumerateTasks();
+                break;
+            case DONE:
+                tasks.markAsDone(input);
+                break;
+            case TODO:
+                parser.checkValidInput(input, c);
+                ui.plusNumOfGoals();
+                tasks.addNewTodo(input);
+                tasks.repeatTaskAdded();
+                break;
+            case EVENT:
+                parser.checkValidInput(input, c);
+                ui.plusNumOfGoals();
+                tasks.addNewEvent(input);
+                tasks.repeatTaskAdded();
+                break;
+            case DEADLINE:
+                parser.checkValidInput(input, c);
+                ui.plusNumOfGoals();
+                tasks.addNewDeadline(input);
+                tasks.repeatTaskAdded();
+                break;
+            case DELETE:
+                tasks.deleteTask(user_input);
+                break;
+            case FIND:
+                tasks.findTasks(user_input);
+                break;
+
+            case INVALID:
+                throw new InvalidCommandException();
```

Comment:
Well done in ensuring [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) is consistent among the cases.